test/node/extended_json.test.ts

```diff
@@ -180,6 +180,30 @@ describe('Extended JSON', function () {
     expect(EJSON.parse(serialized)).to.deep.equal(numbers);
   });
 
+  it('truncates bigint values when they are outside the range [BSON_INT64_MIN, BSON_INT64_MAX] in canonical mode', function () {
+    const numbers = { a: 2n ** 64n + 1n, b: -(2n ** 64n) - 1n };
+    const serialized = EJSON.stringify(numbers, { relaxed: false });
+    expect(serialized).to.equal('{"a":{"$numberLong":"1"},"b":{"$numberLong":"-1"}}');
+  });
+
+  it('serializes bigint values to numberLong in canonical mode', function () {
+    const number = { a: 2n };
+    const serialized = EJSON.stringify(number, { relaxed: false });
+    expect(serialized).to.equal('{"a":{"$numberLong":"2"}}');
+  });
+
+  it('serializes bigint values to Number in relaxed mode', function () {
+    const number = { a: 10000n };
+    const serialized = EJSON.stringify(number, { relaxed: true });
+    expect(serialized).to.equal('{"a":10000}');
+  });
+
+  it('loses precision when serializing bigint values outside of range [Number.MIN_SAFE_INTEGER, Number.MAX_SAFE_INTEGER] in relaxed mode', function () {
+    const numbers = { a: -(2n ** 53n) - 1n, b: 2n ** 53n + 2n };
+    const serialized = EJSON.stringify(numbers, { relaxed: true });
+    expect(serialized).to.equal('{"a":-9007199254740992,"b":9007199254740994}');
+  });
+
```

Comment:
The truncation being the same in BSON and EJSON is an "interesting" case, can we add a test that runs serialize and stringify and we get the "same" results. And same here will mean same in value interpretation, one is going to be bytes and the other will be a string. 
Turning the bytes back into a bigint within the test and checking its stringified value should result in the same string EJSON emits.